modules/ting_reference/js/ting_reference_ajax.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file
+ * Ajax/lazy load ting reference objects into the current page.
+ */
+(function ($) {
+  "use strict";
+
+  Drupal.behaviors.ting_reference_ajax = {
+    attach: function (context) {
+      $('.ting-reference-item', context).once('ting-reference-item', function () {
+        var elm = $(this);
+        var entity_id = elm.data('entity-id');
+        var view_mode = elm.data('view-mode');
+        $.ajax({
+          type: 'get',
+          url : Drupal.settings.basePath + 'ting_reference/ajax/' + entity_id + '/' + view_mode,
+          dataType : 'json',
+          success : function (data) {
+            elm.html(data.content);
+
+            // This ensures that ting objects loaded via ajax in the carousel's gets
+            // reservations buttons displayed if available. So basically it finds
+            // the material ids and coverts them into ding_availability format and
+            // updates the settings, which is this used when behaviors are attached
+            // below. This is a hack, but the alternative was to re-write
+            // ding_availability.
```

Comment:
... and now we are repeating the code 😕 .

Reply:
Yes, we are as the comment says to avoid this ding_availability has to be rewritten to get around not having to do this and we don't have a ding_js_utils library.